subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/coreTypes/SimpleCoreHandlerTest.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.persistence.typeHandling.coreTypes;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.terasology.persistence.typeHandling.PersistedData;
+import org.terasology.persistence.typeHandling.TypeHandler;
+import org.terasology.persistence.typeHandling.inMemory.InMemoryPersistedDataSerializer;
+import org.terasology.persistence.typeHandling.inMemory.PersistedBoolean;
+import org.terasology.persistence.typeHandling.inMemory.PersistedDouble;
+import org.terasology.persistence.typeHandling.inMemory.PersistedFloat;
+import org.terasology.persistence.typeHandling.inMemory.PersistedInteger;
+import org.terasology.persistence.typeHandling.inMemory.PersistedLong;
+import org.terasology.persistence.typeHandling.inMemory.PersistedString;
+
+import java.lang.reflect.Field;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+class SimpleCoreHandlerTest {
+
+    private static Stream<Arguments> primitives() {
+        return Stream.of(
+                Arguments.of(true, new BooleanTypeHandler(), new PersistedBoolean(true)),
+
+                Arguments.of(1.0, new NumberTypeHandler(), new PersistedDouble(1)),
+                Arguments.of(1, new IntTypeHandler(), new PersistedInteger(1)),
+                Arguments.of(1L, new LongTypeHandler(), new PersistedLong(1)),
+                Arguments.of(1.0F, new FloatTypeHandler(), new PersistedFloat(1)),
+                Arguments.of(1.0, new DoubleTypeHandler(), new PersistedDouble(1)),
+
+                Arguments.of("foo", new StringTypeHandler(), new PersistedString("foo")),
+                Arguments.of('f', new CharacterTypeHandler(), new PersistedString("f")));
```

Comment:
Testing some edge cases would be nice, but the setup is quite nice and we can easily add them later.

Reply:
this typehandlers almost haven't edge cases. (related to java's data types)

yeah, I known one case :D sends emoji character to CharacterTypeHandler.. and deserialize it back.